### ISSUE TITLE
How to implement the "RequestedToCapacityRatio" Priorities on the v1.16 k8s server?

### ISSUE THREAD

**Issue by Wenshiqi222** [kubernetes/kubernetes#98414]
Since the k8s is v1.16,i cannot realize the RequestedToCapacityRatio Score Plugins instead of the LeastRequestedPriority by the Framework Plugins; So how can i deal with the problem in v1.16 environment?

**Comment by damemi** [comment]
Hi @Wenshiqi222,
The `RequestedToCapacityRatioPriority` should be available in v1.16, are you trying to set it through a Policy file or the newer scheduler Plugin Config?

**Comment by daDangminh** [comment]
@damemi Hi, I am currently facing the same issue with trying enable RequestedToCapacityRatio plugin in K8s 1.16. What I attempted to do is writing a KubeSchedulerConfiguration YAML file at `/etc/kubernetes/scheduler-config.yaml`, which I then refer to inside the YAML file defining the scheduler pod at `/etc/kubernetes/manifests/kube-scheduler.yaml`.

Here's the content of `/etc/kubernetes/scheduler-config.yaml`.
`apiVersion: kubescheduler.config.k8s.io/v1alpha1
kind: KubeSchedulerConfiguration
leaderElection:
  leaderElect: false
clientConnection:
  kubeconfig: "/etc/kubernetes/scheduler.conf"
plugins:
  score:
    disabled:
    - name: NodeResourcesLeastAllocated
    enabled:
    - name: RequestedToCapacityRatio
      weight: 100
pluginConfig:
- name: RequestedToCapacityRatio
  args:
    #functionshape:
    shape:
      - utilization: 0
        score: 0
      - utilization: 100
        score: 10
    #resourcetoweightmap:
    resources:
      - name: memory
        weight: 1
      - name: nvidia.com/gpu
        weight: 1
`

Meanwhile the content of `/etc/kubernetes/manifests/kube-scheduler.yaml` is as follows:
`apiVersion: v1
kind: Pod
metadata:
  creationTimestamp: null
  labels:
    component: kube-scheduler
    tier: control-plane
  name: kube-scheduler
  namespace: kube-system
spec:
  containers:
  - command:
    - kube-scheduler
    - --authentication-kubeconfig=/etc/kubernetes/scheduler.conf
    - --authorization-kubeconfig=/etc/kubernetes/scheduler.conf
    - --bind-address=127.0.0.1
    - --kubeconfig=/etc/kubernetes/scheduler.conf
    - --leader-elect=true
    - -v=5
    - --config=/etc/kubernetes/scheduler-config.yaml
    image: k8s.gcr.io/kube-scheduler:v1.16.15
    imagePullPolicy: IfNotPresent
    livenessProbe:
      failureThreshold: 8
      httpGet:
        host: 127.0.0.1
        path: /healthz
        port: 10251
        scheme: HTTP
      initialDelaySeconds: 15
      timeoutSeconds: 15
    name: kube-scheduler
    resources:
      requests:
        cpu: 100m
    volumeMounts:
    - mountPath: /etc/kubernetes/scheduler.conf
      name: kubeconfig
      readOnly: true
    - mountPath: /etc/kubernetes/scheduler-config.yaml
      name: customconfig
      readOnly: true
  hostNetwork: true
  priorityClassName: system-cluster-critical
  volumes:
  - hostPath:
      path: /etc/kubernetes/scheduler.conf
      type: FileOrCreate
    name: kubeconfig
  - hostPath:
      path: /etc/kubernetes/scheduler-config.yaml
      type: FileOrCreate
    name: customconfig
status: {}`


Here's the error logs, it clearly say RequestedToCapacityRatio does not exist, even though I am using v1.16.15.
`
I0802 02:33:45.896551       1 flags.go:33] FLAG: --add-dir-header="false"
I0802 02:33:45.896584       1 flags.go:33] FLAG: --address="0.0.0.0"
I0802 02:33:45.896589       1 flags.go:33] FLAG: --algorithm-provider=""
I0802 02:33:45.896592       1 flags.go:33] FLAG: --alsologtostderr="false"
I0802 02:33:45.896596       1 flags.go:33] FLAG: --authentication-kubeconfig="/etc/kubernetes/scheduler.conf"
I0802 02:33:45.896600       1 flags.go:33] FLAG: --authentication-skip-lookup="false"
I0802 02:33:45.896604       1 flags.go:33] FLAG: --authentication-token-webhook-cache-ttl="10s"
I0802 02:33:45.896608       1 flags.go:33] FLAG: --authentication-tolerate-lookup-failure="true"
I0802 02:33:45.896611       1 flags.go:33] FLAG: --authorization-always-allow-paths="[/healthz]"
I0802 02:33:45.896626       1 flags.go:33] FLAG: --authorization-kubeconfig="/etc/kubernetes/scheduler.conf"
I0802 02:33:45.896631       1 flags.go:33] FLAG: --authorization-webhook-cache-authorized-ttl="10s"
I0802 02:33:45.896634       1 flags.go:33] FLAG: --authorization-webhook-cache-unauthorized-ttl="10s"
I0802 02:33:45.896637       1 flags.go:33] FLAG: --bind-address="127.0.0.1"
I0802 02:33:45.896641       1 flags.go:33] FLAG: --cert-dir=""
I0802 02:33:45.896644       1 flags.go:33] FLAG: --client-ca-file=""
I0802 02:33:45.896646       1 flags.go:33] FLAG: --config="/etc/kubernetes/scheduler-config.yaml"
I0802 02:33:45.896651       1 flags.go:33] FLAG: --contention-profiling="false"
I0802 02:33:45.896654       1 flags.go:33] FLAG: --feature-gates=""
I0802 02:33:45.896660       1 flags.go:33] FLAG: --hard-pod-affinity-symmetric-weight="1"
I0802 02:33:45.896664       1 flags.go:33] FLAG: --help="false"
I0802 02:33:45.896667       1 flags.go:33] FLAG: --http2-max-streams-per-connection="0"
I0802 02:33:45.896671       1 flags.go:33] FLAG: --kube-api-burst="100"
I0802 02:33:45.896675       1 flags.go:33] FLAG: --kube-api-content-type="application/vnd.kubernetes.protobuf"
I0802 02:33:45.896679       1 flags.go:33] FLAG: --kube-api-qps="50"
I0802 02:33:45.896687       1 flags.go:33] FLAG: --kubeconfig="/etc/kubernetes/scheduler.conf"
I0802 02:33:45.896693       1 flags.go:33] FLAG: --leader-elect="true"
I0802 02:33:45.896696       1 flags.go:33] FLAG: --leader-elect-lease-duration="15s"
I0802 02:33:45.896703       1 flags.go:33] FLAG: --leader-elect-renew-deadline="10s"
I0802 02:33:45.896706       1 flags.go:33] FLAG: --leader-elect-resource-lock="endpoints"
I0802 02:33:45.896709       1 flags.go:33] FLAG: --leader-elect-resource-name="kube-scheduler"
I0802 02:33:45.896712       1 flags.go:33] FLAG: --leader-elect-resource-namespace="kube-system"
I0802 02:33:45.896716       1 flags.go:33] FLAG: --leader-elect-retry-period="2s"
I0802 02:33:45.896719       1 flags.go:33] FLAG: --lock-object-name="kube-scheduler"
I0802 02:33:45.896723       1 flags.go:33] FLAG: --lock-object-namespace="kube-system"
I0802 02:33:45.896727       1 flags.go:33] FLAG: --log-backtrace-at=":0"
I0802 02:33:45.896735       1 flags.go:33] FLAG: --log-dir=""
I0802 02:33:45.896739       1 flags.go:33] FLAG: --log-file=""
I0802 02:33:45.896747       1 flags.go:33] FLAG: --log-file-max-size="1800"
I0802 02:33:45.896752       1 flags.go:33] FLAG: --log-flush-frequency="5s"
I0802 02:33:45.896755       1 flags.go:33] FLAG: --logtostderr="true"
I0802 02:33:45.896759       1 flags.go:33] FLAG: --master=""
I0802 02:33:45.896762       1 flags.go:33] FLAG: --policy-config-file=""
I0802 02:33:45.896765       1 flags.go:33] FLAG: --policy-configmap=""
I0802 02:33:45.896768       1 flags.go:33] FLAG: --policy-configmap-namespace="kube-system"
I0802 02:33:45.896771       1 flags.go:33] FLAG: --port="10251"
I0802 02:33:45.896775       1 flags.go:33] FLAG: --profiling="false"
I0802 02:33:45.896778       1 flags.go:33] FLAG: --requestheader-allowed-names="[]"
I0802 02:33:45.896785       1 flags.go:33] FLAG: --requestheader-client-ca-file=""
I0802 02:33:45.896790       1 flags.go:33] FLAG: --requestheader-extra-headers-prefix="[x-remote-extra-]"
I0802 02:33:45.896794       1 flags.go:33] FLAG: --requestheader-group-headers="[x-remote-group]"
I0802 02:33:45.896800       1 flags.go:33] FLAG: --requestheader-username-headers="[x-remote-user]"
I0802 02:33:45.896804       1 flags.go:33] FLAG: --scheduler-name="default-scheduler"
I0802 02:33:45.896808       1 flags.go:33] FLAG: --secure-port="10259"
I0802 02:33:45.896811       1 flags.go:33] FLAG: --skip-headers="false"
I0802 02:33:45.896814       1 flags.go:33] FLAG: --skip-log-headers="false"
I0802 02:33:45.896817       1 flags.go:33] FLAG: --stderrthreshold="2"
I0802 02:33:45.896820       1 flags.go:33] FLAG: --tls-cert-file=""
I0802 02:33:45.896823       1 flags.go:33] FLAG: --tls-cipher-suites="[]"
I0802 02:33:45.896831       1 flags.go:33] FLAG: --tls-min-version=""
I0802 02:33:45.896834       1 flags.go:33] FLAG: --tls-private-key-file=""
I0802 02:33:45.896839       1 flags.go:33] FLAG: --tls-sni-cert-key="[]"
I0802 02:33:45.896846       1 flags.go:33] FLAG: --use-legacy-policy-config="false"
I0802 02:33:45.896849       1 flags.go:33] FLAG: --v="5"
I0802 02:33:45.896852       1 flags.go:33] FLAG: --version="false"
I0802 02:33:45.896859       1 flags.go:33] FLAG: --vmodule=""
I0802 02:33:45.896862       1 flags.go:33] FLAG: --write-config-to=""
I0802 02:33:46.187489       1 serving.go:319] Generated self-signed cert in-memory
I0802 02:33:46.519577       1 server.go:148] Version: v1.16.15
I0802 02:33:46.519720       1 defaults.go:91] TaintNodesByCondition is enabled, PodToleratesNodeTaints predicate is mandatory
I0802 02:33:46.519743       1 server.go:167] Starting Kubernetes Scheduler version v1.16.15
F0802 02:33:46.534690       1 factory.go:227] error initializing the scheduling framework: score plugin "RequestedToCapacityRatio" does not exist
`

Am I doing anything wrong? I don't understand why it says the plugin does not exist because according to the documentation [here](https://kubernetes.io/docs/concepts/scheduling-eviction/resource-bin-packing/), it looks like this feature should be available in K8s 1.16. Can anyone please help me?

**Comment by daDangminh** [comment]
@damemi Hi, I am currently facing the same issue with trying enable RequestedToCapacityRatio plugin in K8s 1.16. What I attempted to do is writing a KubeSchedulerConfiguration YAML file at `/etc/kubernetes/scheduler-config.yaml`, which I then refer to inside the YAML file defining the scheduler pod at `/etc/kubernetes/manifests/kube-scheduler.yaml`.

Here's the content of `/etc/kubernetes/scheduler-config.yaml`.
```
apiVersion: kubescheduler.config.k8s.io/v1alpha1
kind: KubeSchedulerConfiguration
leaderElection:
  leaderElect: false
clientConnection:
  kubeconfig: "/etc/kubernetes/scheduler.conf"
plugins:
  score:
    disabled:
    - name: NodeResourcesLeastAllocated
    enabled:
    - name: RequestedToCapacityRatio
      weight: 100
pluginConfig:
- name: RequestedToCapacityRatio
  args:
    #functionshape:
    shape:
      - utilization: 0
        score: 0
      - utilization: 100
        score: 10
    #resourcetoweightmap:
    resources:
      - name: memory
        weight: 1
      - name: nvidia.com/gpu
        weight: 1
```

Meanwhile the content of `/etc/kubernetes/manifests/kube-scheduler.yaml` is as follows:
```
apiVersion: v1
kind: Pod
metadata:
  creationTimestamp: null
  labels:
    component: kube-scheduler
    tier: control-plane
  name: kube-scheduler
  namespace: kube-system
spec:
  containers:
  - command:
    - kube-scheduler
    - --authentication-kubeconfig=/etc/kubernetes/scheduler.conf
    - --authorization-kubeconfig=/etc/kubernetes/scheduler.conf
    - --bind-address=127.0.0.1
    - --kubeconfig=/etc/kubernetes/scheduler.conf
    - --leader-elect=true
    - -v=5
    - --config=/etc/kubernetes/scheduler-config.yaml
    image: k8s.gcr.io/kube-scheduler:v1.16.15
    imagePullPolicy: IfNotPresent
    livenessProbe:
      failureThreshold: 8
      httpGet:
        host: 127.0.0.1
        path: /healthz
        port: 10251
        scheme: HTTP
      initialDelaySeconds: 15
      timeoutSeconds: 15
    name: kube-scheduler
    resources:
      requests:
        cpu: 100m
    volumeMounts:
    - mountPath: /etc/kubernetes/scheduler.conf
      name: kubeconfig
      readOnly: true
    - mountPath: /etc/kubernetes/scheduler-config.yaml
      name: customconfig
      readOnly: true
  hostNetwork: true
  priorityClassName: system-cluster-critical
  volumes:
  - hostPath:
      path: /etc/kubernetes/scheduler.conf
      type: FileOrCreate
    name: kubeconfig
  - hostPath:
      path: /etc/kubernetes/scheduler-config.yaml
      type: FileOrCreate
    name: customconfig
status: {}
```


Here's the error logs, it clearly say RequestedToCapacityRatio does not exist, even though I am using v1.16.15.
```
I0802 02:33:45.896551       1 flags.go:33] FLAG: --add-dir-header="false"
I0802 02:33:45.896584       1 flags.go:33] FLAG: --address="0.0.0.0"
I0802 02:33:45.896589       1 flags.go:33] FLAG: --algorithm-provider=""
I0802 02:33:45.896592       1 flags.go:33] FLAG: --alsologtostderr="false"
I0802 02:33:45.896596       1 flags.go:33] FLAG: --authentication-kubeconfig="/etc/kubernetes/scheduler.conf"
I0802 02:33:45.896600       1 flags.go:33] FLAG: --authentication-skip-lookup="false"
I0802 02:33:45.896604       1 flags.go:33] FLAG: --authentication-token-webhook-cache-ttl="10s"
I0802 02:33:45.896608       1 flags.go:33] FLAG: --authentication-tolerate-lookup-failure="true"
I0802 02:33:45.896611       1 flags.go:33] FLAG: --authorization-always-allow-paths="[/healthz]"
I0802 02:33:45.896626       1 flags.go:33] FLAG: --authorization-kubeconfig="/etc/kubernetes/scheduler.conf"
I0802 02:33:45.896631       1 flags.go:33] FLAG: --authorization-webhook-cache-authorized-ttl="10s"
I0802 02:33:45.896634       1 flags.go:33] FLAG: --authorization-webhook-cache-unauthorized-ttl="10s"
I0802 02:33:45.896637       1 flags.go:33] FLAG: --bind-address="127.0.0.1"
I0802 02:33:45.896641       1 flags.go:33] FLAG: --cert-dir=""
I0802 02:33:45.896644       1 flags.go:33] FLAG: --client-ca-file=""
I0802 02:33:45.896646       1 flags.go:33] FLAG: --config="/etc/kubernetes/scheduler-config.yaml"
I0802 02:33:45.896651       1 flags.go:33] FLAG: --contention-profiling="false"
I0802 02:33:45.896654       1 flags.go:33] FLAG: --feature-gates=""
I0802 02:33:45.896660       1 flags.go:33] FLAG: --hard-pod-affinity-symmetric-weight="1"
I0802 02:33:45.896664       1 flags.go:33] FLAG: --help="false"
I0802 02:33:45.896667       1 flags.go:33] FLAG: --http2-max-streams-per-connection="0"
I0802 02:33:45.896671       1 flags.go:33] FLAG: --kube-api-burst="100"
I0802 02:33:45.896675       1 flags.go:33] FLAG: --kube-api-content-type="application/vnd.kubernetes.protobuf"
I0802 02:33:45.896679       1 flags.go:33] FLAG: --kube-api-qps="50"
I0802 02:33:45.896687       1 flags.go:33] FLAG: --kubeconfig="/etc/kubernetes/scheduler.conf"
I0802 02:33:45.896693       1 flags.go:33] FLAG: --leader-elect="true"
I0802 02:33:45.896696       1 flags.go:33] FLAG: --leader-elect-lease-duration="15s"
I0802 02:33:45.896703       1 flags.go:33] FLAG: --leader-elect-renew-deadline="10s"
I0802 02:33:45.896706       1 flags.go:33] FLAG: --leader-elect-resource-lock="endpoints"
I0802 02:33:45.896709       1 flags.go:33] FLAG: --leader-elect-resource-name="kube-scheduler"
I0802 02:33:45.896712       1 flags.go:33] FLAG: --leader-elect-resource-namespace="kube-system"
I0802 02:33:45.896716       1 flags.go:33] FLAG: --leader-elect-retry-period="2s"
I0802 02:33:45.896719       1 flags.go:33] FLAG: --lock-object-name="kube-scheduler"
I0802 02:33:45.896723       1 flags.go:33] FLAG: --lock-object-namespace="kube-system"
I0802 02:33:45.896727       1 flags.go:33] FLAG: --log-backtrace-at=":0"
I0802 02:33:45.896735       1 flags.go:33] FLAG: --log-dir=""
I0802 02:33:45.896739       1 flags.go:33] FLAG: --log-file=""
I0802 02:33:45.896747       1 flags.go:33] FLAG: --log-file-max-size="1800"
I0802 02:33:45.896752       1 flags.go:33] FLAG: --log-flush-frequency="5s"
I0802 02:33:45.896755       1 flags.go:33] FLAG: --logtostderr="true"
I0802 02:33:45.896759       1 flags.go:33] FLAG: --master=""
I0802 02:33:45.896762       1 flags.go:33] FLAG: --policy-config-file=""
I0802 02:33:45.896765       1 flags.go:33] FLAG: --policy-configmap=""
I0802 02:33:45.896768       1 flags.go:33] FLAG: --policy-configmap-namespace="kube-system"
I0802 02:33:45.896771       1 flags.go:33] FLAG: --port="10251"
I0802 02:33:45.896775       1 flags.go:33] FLAG: --profiling="false"
I0802 02:33:45.896778       1 flags.go:33] FLAG: --requestheader-allowed-names="[]"
I0802 02:33:45.896785       1 flags.go:33] FLAG: --requestheader-client-ca-file=""
I0802 02:33:45.896790       1 flags.go:33] FLAG: --requestheader-extra-headers-prefix="[x-remote-extra-]"
I0802 02:33:45.896794       1 flags.go:33] FLAG: --requestheader-group-headers="[x-remote-group]"
I0802 02:33:45.896800       1 flags.go:33] FLAG: --requestheader-username-headers="[x-remote-user]"
I0802 02:33:45.896804       1 flags.go:33] FLAG: --scheduler-name="default-scheduler"
I0802 02:33:45.896808       1 flags.go:33] FLAG: --secure-port="10259"
I0802 02:33:45.896811       1 flags.go:33] FLAG: --skip-headers="false"
I0802 02:33:45.896814       1 flags.go:33] FLAG: --skip-log-headers="false"
I0802 02:33:45.896817       1 flags.go:33] FLAG: --stderrthreshold="2"
I0802 02:33:45.896820       1 flags.go:33] FLAG: --tls-cert-file=""
I0802 02:33:45.896823       1 flags.go:33] FLAG: --tls-cipher-suites="[]"
I0802 02:33:45.896831       1 flags.go:33] FLAG: --tls-min-version=""
I0802 02:33:45.896834       1 flags.go:33] FLAG: --tls-private-key-file=""
I0802 02:33:45.896839       1 flags.go:33] FLAG: --tls-sni-cert-key="[]"
I0802 02:33:45.896846       1 flags.go:33] FLAG: --use-legacy-policy-config="false"
I0802 02:33:45.896849       1 flags.go:33] FLAG: --v="5"
I0802 02:33:45.896852       1 flags.go:33] FLAG: --version="false"
I0802 02:33:45.896859       1 flags.go:33] FLAG: --vmodule=""
I0802 02:33:45.896862       1 flags.go:33] FLAG: --write-config-to=""
I0802 02:33:46.187489       1 serving.go:319] Generated self-signed cert in-memory
I0802 02:33:46.519577       1 server.go:148] Version: v1.16.15
I0802 02:33:46.519720       1 defaults.go:91] TaintNodesByCondition is enabled, PodToleratesNodeTaints predicate is mandatory
I0802 02:33:46.519743       1 server.go:167] Starting Kubernetes Scheduler version v1.16.15
F0802 02:33:46.534690       1 factory.go:227] error initializing the scheduling framework: score plugin "RequestedToCapacityRatio" does not exist
```

Am I doing anything wrong? I don't understand why it says the plugin does not exist because according to the documentation [here](https://kubernetes.io/docs/concepts/scheduling-eviction/resource-bin-packing/), it looks like this feature should be available in K8s 1.16. Can anyone please help me?

**Comment by damemi** [comment]
It might be the case that the plugin name is actually `RequestedToCapacityRatioResourceAllocation`, though the yaml example only shows `RequestedToCapacityRatio`. If that's so, then we should update this docs page
